sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -556,6 +556,109 @@ Parameters:
 
 |===
 
+==== Dynamic Command Groups
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+class dynamic_command_group {
+public:
+  dynamic_command_group(
+      command_graph<graph_state::modifiable> graph, const std::vector<std::function<void(handler &)>>& cgfList);
+
+  size_t get_active_cgf();
+  void set_active_cgf(size_t cgfIndex);
+};
+----
+
+Dynamic command-groups can be added as nodes to a graph. They provide a mechanism that
+allows updating the command-group function of a node after the graph is finalized.
+There is always one command-group function in the dynamic command-group that is set
+as active. When a dynamic command-group node is executed, the kernel of the active
+command-group function will be run and all the other command-group functions in
+`cgfList` will be ignored.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating command-groups.
+
+===== Limitations
+
+Dynamic command-groups can only be used to update kernels. Trying to update a command group
+function that contains other operations will result in an error.
+
+All the command-group functions in a dynamic command-group must have identical dependencies.
+It is not allowed for a dynamic command-group to have command-group functions that would
```

Comment:
"It is not allowed" - Should we specify and error for this, or does it result in UB?

Reply:
I think we should specify an error. Ideally I would like that error to be on dynamic command-group construction. But I'm not sure how easy it is to check that. Depends on the other discussion about error checking: https://github.com/reble/llvm/pull/378/files#r1687772639

Otherwise, we can check for this during node add() or during a call to update().

Reply:
Wonder if we'll need to finalize a handler to get this information, unlike for that other thread. Like I suggested there, we could specify our ideal behavior for now then refine it when we try to implement it.

Reply:
Added this error definition to the dynamic command group constructor exceptions.